### DISCUSSION TITLE
At least 3-4 seconds of latency using HLL-LL

### DISCUSSION THREAD

**Discussion by dbuscaino** [bluenviron/mediamtx#1065] [Questions]
Hi, I use ffmpeg to send an rtsp stream from raspberry to rtsp-simple-server running on aws. From a web page I use an iframe tag to play the stream using HLS-LL, as indicated in the README.md. I also tried some HLS players, including the one recommended in the README.md, but the video latency is always over 3-4 seconds. If I try to play the rtsp stream using opencv in python (always through rtsp-simple-server) the latency is less than 1 second. I think the high latency is caused by a bad configuration between rtsp stream and HLS related parameters in the rtsp simple server configuration file.

Here my ffmpeg command:
`ffmpeg -input_format h264 -f video4linux2 -video_size 800x600 -framerate 30 -i /dev/video0 -c:v copy -an -preset ultrafast -tune zerolatency -r udp -f rtsp rtsp_server`

Here my HLS part of the rtsp-simple-server configuration file:
```

hlsAlwaysRemux: yes

hlsVariant: lowLatency

hlsSegmentCount: 10

hlsSegmentDuration: 1s

hlsPartDuration: 1s

hlsSegmentMaxSize: 50M
```

RESOLVED: I think the problem was the number of core available on aws. Running the same configuration in my local pc everything works correctly.

**Comment by aler9** [comment]
Hello, glad you solved the issue, but i can assure you that the most important parameter that allows to decrease latency is `hlsSegmentDuration`, the less it is, the less the latency (at cost of a huge quantity of requests / responses to / from the server):

```yml
hlsSegmentDuration: 100ms
```

[a comment by an automated account is not shown]
